### ISSUE TITLE
constify more of the GSS-API

### ISSUE THREAD

**Issue by nicowilliams** [heimdal/heimdal#501]
During an OpenJDK code review it was noted that neither MIT nor Heimdal use the various `gss_const_*` types in pre-`RFC5587` functions.  We should make that change.

**Comment by lhoward** [comment]
Yeah, and I'd be careful const-ing more stuff, first I think we were a bit zealous (e.g. const input cred in `gss_add_cred()`), and secondly, the last round of changes broke third-party mechanisms that implement API-as-SPI. We fixed it in mech_eap, but it's a pain to support multiple versions with different const'ness. Easier for applications.

**Comment by nicowilliams** [comment]
To clarify, constification of gssapi.h should be backwards-compatible for applications, but source- and ABI-wise.  It does break source compatibility for out-of-tree API-as-SPI mechanisms.  I'm ok with that.
